enhancements/monitoring/monitoring-stack-operator.md

```diff
@@ -0,0 +1,490 @@
+* SaaS
+* A combination of the first two (SaaS that requires an Addon)
+
+The way these managed services are intended to run is presented in the following diagram.
+
+![drawing](https://docs.google.com/drawings/d/e/2PACX-1vQrMBfwE8DJm3kVtC6SLVYJLEztJRABIYwvILMgtOmyyE6WqKSCqvZHksUsVUoqSZIt5E_4ro4ObGmY/pub?w=1312&h=511)
+
+We expect many more Addons to come in the following years.
+
+## Motivation
+
+Red Hat managed services are deployed in OSD clusters and run alongside user workloads.
+
+Even though there are two monitoring stacks available in OSD clusters, neither is suitable for managed services. In OSD, Platform Monitoring is exclusively owned and managed by Platform SRE. It is customized to their needs and no modifications are allowed.
+Because of this, managed tenant owners (MTOs) cannot create additional ServiceMonitors, Prometheus rules and other monitoring resources on metrics in the Platform Monitoring domain.
+
+User Workload Monitoring was designed to be primarily used by customers for monitoring their own workloads. Managed services are discouraged from using this stack since they could easily get DOS-ed by high cardinality data from end-users.
+Similarly, the configuration of the Prometheus instance itself and Alerting and Routing, would be shared with customers and other managed service owners.
+
+As a result, each MTOs currently needs to deploy its own self-developed monitoring solutions alongside its service.
+Current options like [Prometheus Operator](https://operatorhub.io/operator/prometheus) (PO) through OLM provide inital value, but they require expertise on how to build a full-fledged monitoring stack with multi-tenancy and high availability in mind (RBAC proxies, HA, Thanos Querier, Ruler etc.).
+This imposes an extra cost on teams developing and maintaining a managed service and increases time-to-market for new services.
+
+### Goals
+
+This enhancement aims to relieve the burden from managed service owners by providing a ready-made monitoring solution. Such a solution would encapsulate existing patterns already present in Platform Monitoring and User Workload monitoring.
+Furthermore, it will reduce the operational complexity of running a managed service since the Managed Tenant SREs will not need to familiarise themselves with a new monitoring stack for each managed service.
+In future it can be used to allow customers to define their monitoring stacks. UWM could use it in future underneath too. In details:
+
+* Provide a way for managed services to scrape metrics from their own services and from platform components into prometheus-compatible storage.
+* Allow managed service owners to query metrics from their services and platform components from prometheus-compatible storage.
+* Provide a way for managed services to define alerting and recording rules on their own metrics and a subset of platform metrics.
+* Provide a way for managed services to create dashboards for monitoring their services.
```

Comment:
Are there plans to provide OCP console integration? Currently the console queries thanos in cluster that provides both openshift-monitoring + openshift-user-workload-monitoring alerts for the console to display. I am unsure if this is desirable, just raising the point.